### DISCUSSION TITLE
Getting Errors after Cloning Project from Repository

### DISCUSSION THREAD

**Discussion by MosesAlfred** [community/community#27268] [New to GitHub]
I have cloned a Project from GitHub After cloning Project i have opened it In visual studio 2019 i am getting Changes to commit. But Actually i did not do any changes  i have just open project 
![image](https://user-images.githubusercontent.com/79455893/181207368-8a4e932f-479b-452f-b9ee-6ff5f08de2fb.png)

**Comment by airtower-luna** [comment]
Those look like files Visual Studio generated while importing and building the project. You should add them (or maybe better appropriate patterns) to `.gitignore`, or (if you want to keep the ignore rules local-only) to `.git/info/exclude`. That way Git won't try to add the files.

See also: https://git-scm.com/docs/gitignore

**Reply by JSOUSA90** [reply]
You can also do `git status` to display the state of the working directory and the staging area.

**Reply by airtower-luna** [reply]
That is correct, and specifically selecting files to commit with `git add` (and `git reset` to undo) is a good idea anyway. But I've seen *a lot* of cases where people accidentally committed files they didn't mean to, and putting them into `.gitignore` prevents that reliably. :smiley_cat:

**Comment by eragon11** [comment]
I think .git head not detected in your local repo. Please reset head and add some unwanted things in .gitignore and try to commit . It will solve your problem.

**Comment by MosesAlfred** [comment]
Hi @airtower-luna @JSOUSA90 @eragon11 
Thanks For Your Reply's

Actually I have Imported this Project from TFS.
Here I did not have .gitignore file
Please help me with some more Information to overcome this issue

There are the files I got to GitHub after Importing From TFS
![image](https://user-images.githubusercontent.com/79455893/181419724-c3d4ea44-c1cc-4090-a805-4df6c851dd59.png)

**Reply by airtower-luna** [reply]
> Here I did not have .gitignore file

That's why I suggested you should add it.

Many software projects will generate files during compilation, and IDEs (in your case Visual Studio) add their own configs. In most cases you don't want those files in the repository, so you use a `.gitignore` file to tell Git to, well, ignore them. :slightly_smiling_face:

As described in the documentation I linked, the `.gitignore` file lists filenames and filename patterns to ignore. It's probably easiest if you create it locally and try with `git status` if the files you want to ignore are actually ignored. Note that already added files can't be ignored, you'd need to reset (if they aren't committed yet) or remove them from the repository.
